com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -93,33 +93,24 @@ static CloudControllerClient getControllerClient(DelegateExecution context, Clou
     }
 
     public static String determineCurrentUser(VariableScope scope, StepLogger stepLogger) {
-        String userId = Authentication.getAuthenticatedUserId();
-        String previousUser = (String) scope.getVariable(Constants.VAR_USER);
-        // Determine the current user
-        if (userId != null && !userId.equals(previousUser)) {
-            stepLogger.debug(Messages.AUTHENTICATED_USER_ID, userId);
-            stepLogger.debug(Messages.PREVIOUS_USER, previousUser);
-        }
+        String userId = (String) scope.getVariable(Constants.VAR_USER);
         if (userId == null) {
-            // If the authenticated user cannot be determined,
-            // use the user saved by the previous service task
-            userId = previousUser;
-            if (userId == null) {
-                // If there is no previous user, this must be the first service task
-                // Use the process initiator in this case
-                userId = (String) scope.getVariable(Constants.PARAM_INITIATOR);
-                stepLogger.debug(Messages.PROCESS_INITIATOR, userId);
-                if (userId == null) {
-                    throw new SLException(Messages.CANT_DETERMINE_CURRENT_USER);
-                }
-            }
+            userId = getProcessInitiator(scope, stepLogger);
+            scope.setVariable(Constants.VAR_USER, userId);
+            stepLogger.debug(Messages.UPDATED_CURRENT_USER, userId);

```

Comment:
Several questions:
1. You already have a debug message inside getProcessInitiator method, and right after that you add this one. They'll be identical, except for a slightly different message, but containing the same userId. This duplication is unnecessary, am I correct?

2. If this is meant to notify on change of the current userId, then I do not think it will do just that. The userId will be updated by the corresponding action, and it will be retrieved on the first line in this method. The branching here is used only if there is no 'userId' as a fallback scenario. Previously to check if the user has changed we had previousUser to compare with. This (current) branching is not handling a normal update of the userId, therefore this message should not be here.